cupyx/scipy/ndimage/measurements.py

```diff
@@ -200,3 +203,282 @@ def _kernel_finalize():
+
+    Returns:
+        variance (cupy.ndarray): Values of variance, for each sub-region if
+        `labels` and `index` are specified.
+
+
+    .. seealso:: :func:`scipy.ndimage.variance`
+    """
+    if not isinstance(input, cupy.ndarray):
+        raise TypeError('input must be cupy.ndarray')
+
+    # There is constraints on types because atomicAdd() in CUDA 7.5
+    # only supports int32, uint32, uint64, and float32.
+    if not issubclass(input.dtype.type,
+                      (cupy.int32, cupy.float16, cupy.float32, cupy.float64,
+                       cupy.uint32, cupy.uint64, cupy.ulonglong)):
+        raise TypeError(
+            'cupyx.scipy.ndimage.variance only supports int32, float16, '
+            'float32, float64, uint32, uint64, as data type')
+
+    if labels is None:
+        return input.var()
+
+    if not isinstance(labels, cupy.ndarray):
+        raise TypeError('label must be cupy.ndarray')
+
+    if index is None:
+        return (input[labels != 0]).var()
+
+    input, labels = cupy.broadcast_arrays(input, labels)
+
+    if not isinstance(index, cupy.ndarray):
+        if not isinstance(index, int):
+            raise TypeError('index must be cupy.ndarray or a scalar int')
+        else:
+            return (input[labels == index]).var()
+
+    mean_val, count = _mean_driver(input, labels, index, True)
+    out = cupy.zeros_like(index, dtype=mean_val.dtype)
+    return _ndimage_variance_kernel(input, labels, index, index.size, mean_val,
+                                    out)/count
+
+
+def sum(input, labels=None, index=None):
+    """Calculate the sum of the values of an n-D image array, optionally
+       at specified sub-regions.
+
+    Args:
+        input (cupy.ndarray) :Nd-image data to process.
+        labels (cupy.ndarray or None): Labels defining sub-regions in `input`.
+            If not None, must be same shape as `input`.
+        index (cupy.ndarray or None): `labels` to include in output.  If None
+            (default), all values where `labels` is non-zero are used.
+
+    Returns:
+       sum (cupy.ndarray): sum of values, for each sub-region if
+       `labels` and `index` are specified.
+
+
+    .. seealso:: :func:`scipy.ndimage.sum`
+    """
+    if not isinstance(input, cupy.ndarray):
+        raise TypeError('input must be cupy.ndarray')
+
+    if issubclass(input.dtype.type, (cupy.bool, cupy.complex64,
+                                     cupy.complex128)):
+        raise TypeError("cupyx.scipy.ndimage.sum doesnt support %{}".format(
+            input.dtype.type))
+
+    use_kern = False
+    # There is constraints on types because atomicAdd() in CUDA 7.5
+    # only supports int32, uint32, uint64, and float32.
+    if not issubclass(input.dtype.type,
+                      (cupy.int32, cupy.float16, cupy.float32,
+                       cupy.float64, cupy.uint32, cupy.uint64,
+                       cupy.ulonglong)):
+        warnings.warn(
```

Comment:
Because of the matching kernel, sum and mean both now support all dtypes other than complex64, complex128 and bool because scipy doesnt support them. If a operation uses this without the performance threshold a warning is thrown currently(should it ? ).

Reply:
I think it is fine to throw this warning as users would think they change the dtype or not.